elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation;
+
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+
+/**
+ * A {@link QueryEngine} is an abstraction layer that {@link DataStoreTransaction} leverages to optimize query and send
+ * query, such as loading objects, to a persistent storage.
+ * <p>
+ * The purpose of {@link QueryEngine} is to allow a single {@link DataStore} to utilize multiple query frameworks, such
+ * as Hibernate on SQL and Drill on NoSQL shown below. There will be 3 implementations of {@link QueryEngine}:
+ * DrillQueryEngine, JpaQueryEngine, and HibernateQueryEngine
+ * <pre>
+ *                              +-----------+
+ *                              |           |
+ *                              | DataStore |
+ *                              |           |
+ *                              +-----+-----+
+ *                                    |
+ *                                    |
+ *                         +----------v-----------+
+ *                         |                      |
+ *                         | DataStoreTransaction |
+ *                         |                      |
+ *                         +----------+-----------+
+ *                                    |
+ *                                    |
+ *                             +------v------+
+ *                             |             |
+ *                             | QueryEngine |
+ *                             |             |
+ *                             +------+------+
+ *                                    |
+ *                                    |
+ *             +-----------------------------------------------+
+ *             |                      |                        |
+ *             |                      |                        |
+ * +-----------v----------+      +----v----+          +--------v-------+
+ * |     Apache Drill     |      |   JPA   |          |    Hibernate   |
+ * +-----------+----------+      +----+----+          +--------+-------+
+ *             |                      |                        |
+ * +-----------v----------------------v------------------------v----------+
+ * |                             SQL Driver                               |
+ * +-------------------+----------------------------+---------------------+
+ *                     |                            |
+ *               +-----v-----+                  +---v----+
+ *               |           |                  |        |
+ *               |  NoSQL DB |                  | SQL DB |
+ *               |           |                  |        |
+ *               +-----------+                  +--------+
```

Comment:
Sorry I made it confusing. Originally they are separate because DrillQueryEngine is a QueryEngine implementation, while Apache Drill is out of our control.

I've updated the diagram; however, if you still don't like it I'm happy to remove it then.